lib/cli-engine.js

```diff
@@ -272,6 +272,30 @@ function createIgnoreResult(filePath, baseDir) {
     };
 }
 
+/**
+ * Produces rule warnings (i.e. deprecation) from configured rules
+ * @param {Object} rules - Rules configured
+ * @param {Map} loadedRules - Map of loaded rules
+ * @returns {Object} Contains rule warnings
+ * @private
+ */
+function createRuleWarnings(rules, loadedRules) {
+    const ruleWarnings = { usedDeprecatedRules: [] };
+
+    if (!rules) {
```

Comment:
I know this isn't a terribly complex function, but I wonder if we could reduce the complexity a bit by restructuring it as:

```
const ruleWarnings = { usedDeprecatedRules: [] };

if (rules) {
    Object.keys(rules).forEach(name => {
        const loadedRule = loadedRules.get(name);

         if (loadedRule.meta && loadedRule.meta.deprecated) {
            ruleWarnings.usedDeprecatedRules.push({ ruleId: name, replacedBy: loadedRule.meta.docs.replacedBy });
        }
    });
}

return ruleWarnings;
```

Reply:
No problem!